aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -47,18 +47,24 @@ Tensor sparse_csr_tensor(
     c10::optional<bool> pin_memory) {
   // See [Note: hacky wrapper removal for TensorOptions]
   TensorOptions options = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
+
+  TORCH_CHECK_NOT_IMPLEMENTED(
+    options.device().type() == kCPU || options.device().type() == kCUDA,
+     "Could not run '", "sparse_csr_tensor", "' from the '", options.device(), "' device.)");
+
   TORCH_CHECK(
       options.layout() == kSparseCsr,
       "expected sparse CSR layout, but got layout ",
       options.layout());
 
   AT_DISPATCH_INDEX_TYPES(crow_indices.scalar_type(), "csr_construct_check", [&] {
-    auto crow_indices_accessor = crow_indices.accessor<index_t, 1>();
+    index_t first_index_value = crow_indices.select(0, 0).item<index_t>();
+    index_t last_index_value = crow_indices.select(0, crow_indices.numel() - 1).item<index_t>();
```

Comment:
If we're assuming `crow_indices` is contiguous (which I'm pretty sure we should), this code is worse because the `select` operation is relatively heavyweight and has to allocate a whole new TensorImpl before you just poke out a single item. The accessor code was better.

Reply:
One approach to support for CUDA requires to have `crow_indices` in CPU (a copy operations CUDA->CPU) to check invariants. @pearu did some quick tests and he figured out that just getting one element is less expensive than  copying to cpu the whole Tensor.
    
https://github.com/pytorch/pytorch/pull/57274#discussion_r624956412
```
>>> start=time.time();b1=a.to(device='cpu'); b=b[0], b[-1]; print(time.time()-start)
0.01662921905517578
>>> start=time.time();b=tuple(a[::a.numel()-1].to(device='cpu')); print(time.time()-start)
0.0002875328063964844
>>> start=time.time();b=a[0].to(device='cpu'),a[-1].to(device='cpu'); print(time.time()-start)
0.013242721557617188
```
What I think is that I can create two paths to do that, the simple one using `accessor` for CPU device and another for CUDA device I will check times using this approach.

Reply:
That is fair. Actually I think this is inextricably linked to the discussion at https://github.com/pytorch/pytorch/pull/57274#discussion_r626829673 so maybe we will know what to do here after we resolve that thread.

Reply:
Solved here: 
https://github.com/pytorch/pytorch/pull/57274#discussion_r626829673